File: templates/ruflet_flutter_template/apple_packages/ruflet_apple/Sources/RufletExtensions/RufletAudio/Sources/Extension.swift
import RufletEngine

@MainActor
public struct RufletAudioExtension: RufletExtension {
  public init() {}

  public func createService(for control: RufletControl) -> RufletService? {
    control.type == "Audio" ? AudioService(control: control) : nil
  }
}

File: templates/ruflet_flutter_template/ruflet_packages/ruflet_audio/lib/src/extension.dart
import 'package:ruflet/ruflet.dart';

import 'audio.dart';

class Extension extends RufletExtension {
  @override
  RufletService? createService(Control control) {
    switch (control.type) {
      case "Audio":
        return AudioService(control: control);
      default:
        return null;
    }
  }
}
